bindings/node/lib/autoEncrypter.js

```diff
@@ -276,3 +284,47 @@ module.exports = function (modules) {
 
   return { AutoEncrypter };
 };
+
+const decryptedKeys = Symbol.for('@@mdb.decryptedKeys');
+/**
+ * Recurse through the (identically-shaped) `decrypted` and `original`
+ * objects and attach a `decryptedKeys` property on each sub-object that
+ * contained encrypted fields. Because we only call this on BSON responses,
+ * we do not need to worry about circular references.
+ */
+function decorateDecryptionResult(decrypted, original, bson, isTopLevelDecorateCall = true) {
+  if (isTopLevelDecorateCall) {
+    // The original value can have been either a JS object or a BSON buffer
+    if (Buffer.isBuffer(original)) {
+      original = bson.deserialize(original);
+    }
+    if (Buffer.isBuffer(decrypted)) {
+      throw new Error('Expected result of decryption to be deserialize BSON object');
```

Comment:
Despite this error being pretty fatal to the process we're trying to do here, as would be bson.deserializing errors, I think we should make sure we funnel any throws here through the callback to `decrypt`.

Reply:
Yeah, makes sense – done! :+1: